wgpu-core/src/resource.rs

```diff
@@ -636,11 +636,17 @@ impl Buffer {
                     });
                     let transition_src = hal::BufferBarrier {
                         buffer: staging_buffer.raw(),
-                        usage: hal::BufferUses::MAP_WRITE..hal::BufferUses::COPY_SRC,
+                        usage: hal::StateTransition {
+                            from: hal::BufferUses::MAP_WRITE,
+                            to: hal::BufferUses::COPY_SRC,
+                        },
                     };
                     let transition_dst = hal::BufferBarrier::<dyn hal::DynBuffer> {
                         buffer: raw_buf,
-                        usage: hal::BufferUses::empty()..hal::BufferUses::COPY_DST,
+                        usage: hal::StateTransition {
+                            from: hal::BufferUses::empty(),
+                            to: hal::BufferUses::COPY_SRC,
+                        },
```

Comment:
issue: Found the bug! 😀

Reply:
Solution: s/SRC/DST in the "new" portion of the diff.

Reply:
Pushed a fix with 8e8d04f74c791f9f39deb29f93385c05dfc25db4.

Reply:
Ah! Good catch!